front/app/modules/commercial/project_folders/admin/components/ProjectFolderSelect/index.tsx

```diff
@@ -119,37 +127,27 @@ const ProjectFolderSelect = memo<Props & InjectedIntlProps>(
       }
     };
 
-    const handleFolderIdChange = useCallback(
-      (folderId: string | null, submitState: 'enabled' | 'disabled') => {
-        if (folderId === null) {
-          setFolderSelected(false);
-        } else {
-          setFolderSelected(true);
-        }
-
-        onChange({
-          submitState,
-          'projectAttributesDiff.folder_id': folderId,
-        });
-      },
-      [onChange]
-    );
+    const handleFolderIdChange = (
+      folderId: string | null,
+      submitState: 'enabled' | 'disabled'
+    ) => {
+      setFolderSelected(folderId ? true : false);
 
-    const onRadioFolderSelectChange = useCallback(
-      (newRadioProjectFolderSelect: boolean) => {
-        if (newRadioProjectFolderSelect === true) {
-          setRadioFolderSelect(true);
-          handleFolderIdChange(null, 'disabled');
-        }
+      onProjectAttributesDiffChange(
+        { folder_id: folderId || undefined },
+        submitState
+      );
+    };
 
-        if (newRadioProjectFolderSelect === false) {
-          setRadioFolderSelect(false);
-          handleFolderIdChange(null, 'enabled');
-        }
-      },
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-      [onChange]
-    );
+    const onRadioFolderSelectChange = (
+      newRadioProjectFolderSelect: boolean
+    ) => {
+      setRadioFolderSelect(newRadioProjectFolderSelect);
+      handleFolderIdChange(
+        null,
+        newRadioProjectFolderSelect ? 'disabled' : 'enabled'
+      );
```

Comment:
Maybe not ideal we automatically deselect a folder if we already had one... Added a comment :-)